idf_build_apps/app.py

```diff
@@ -130,6 +130,9 @@ def __init__(
         build_dir: str = 'build',
         build_log_filename: t.Optional[str] = None,
         size_json_filename: t.Optional[str] = None,
+        size_archives: bool = False,
+        size_archive_details: t.Optional[str] = None,
+        size_files: bool = False,
```

Comment:
Using a unified `size_json_extra_args` instead of separate ones would likely offer more flexibility and simplify the code.